packages/discovery-provider/src/queries/query_helpers.py

```diff
@@ -656,6 +793,18 @@ def getContentId(metadata):
                 "download": has_download_access,
             }
 
+        if entity.get("track_id"):
```

Comment:
this code is so rough. i still think in future we should just have an access table we can hit. and we figure out how to get NFTs to conform to that model

Reply:
Fwiw all that would change (most likely) is the implementation for `check_access_for_batch` though, this particular code would be very similar